addon/components/frost-list-item-content.js

```diff
@@ -0,0 +1,103 @@
+import Ember from 'ember'
+const {get, isPresent} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+
+import layout from '../templates/components/frost-list-item-content'
+
+export default Component.extend({
+
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    // Options - required
+    model: PropTypes.EmberObject.isRequired,
+    index: PropTypes.number.isRequired,
+    onExpand: PropTypes.func.isRequired,
+    onSelect: PropTypes.func.isRequired,
+
+    // Options - general
+    onSelectionChange: PropTypes.func,
+    itemTypes: PropTypes.object,
+    itemTypeKey: PropTypes.string,
```

Comment:
Is this mandatory now?

Reply:
It's not mandatory, but I did add a warning that will show in the console if `itemKeyType` is set and `itemKey` is not set.